submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/wiremock/FireMockDatabase.kt

```diff
@@ -28,55 +30,56 @@ class FireMockDatabase(
     }
 
     fun setPath(fireOid: String, path: String) {
-        if (recordsById.values.firstOrNull { it.path == path } != null)
-            throw FireException("Path '$path' has already a file", BAD_REQUEST)
+        val normalizedPath = normalizePath(path)
+        val firePath = "/$normalizedPath"
+        if (recordsById.values.firstOrNull { it.path == firePath } != null)
+            throw FireException("Path '$firePath' is already allocated", CONFLICT)
 
         val record = recordsById.getValue(fireOid)
-        recordsById[fireOid] = record.copy(path = path)
+        recordsById[fireOid] = record.copy(path = firePath)
```

Comment:
it looks to me that we can simple save normalized path instead of firepath and then we do not need to denormalize in every operation.

Reply:
We still would need to add the extra slash when looking though